### PR TITLE
Cut down some code, fix typos

### PR DESCRIPTION
**Pull request by erasaur**
Largely untested, but no drastic changes. For the `_run` check on `Rules[action][subject]`, is it supposed to default to `passes = true`? It seems like it makes more sense to default to "deny". I will take a look at the other files in more depth when I have time (one thing that immediately strikes me is the `empties` -- what is that for, exactly?).

Also, typo on the `destroyAllRules` :)

### CODE REVIEW

**Comment by Kestanous** [comment]
I love it! As always @erasaur you manage to take my prototypes and DRY the up so elegantly. I will test it shortly.

I keep going back and forth with allow vs deny. My thought is if its not defined default to allow (so we don't interfere with other packages), but we can allow users to whitelist and thus default to deny. 

I am finding the condition to be counter intuitive. The idea is "does this rule apply to this case?" if not it should not count as a defined rule and return based on the whitelist. But I find I put too much logic in it, not sure how to handle that just yet.

As for the `empties` I am embarrassed to admit that I don't know the proper way to turn an array into an object where the array elements are properties and the values are all the `arg`. In this case the `arg` is an array of objects called `empties` for lack of a better name.

As always thanks for all your help. Have you had a chance to use the package? Any thoughts?

**Comment by Kestanous** [comment]
Hmm the tests are all breaking... Will look into it but `Candid: Can.can invalid action "db"`

File: lib/can.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ Can = {
   },
   can: function (action, subject, user, condition) {
 
-    if (dbArray.indexOf(action) == -1) {
+    if (_.contains(dbArray, action)) {
```
Comment (Kestanous) [inline lib/can.js]: Should be: if does not contain or `if (!_.contains(dbArray, action))`

File: lib/can.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ Can = {
       user = user || Meteor.user()
     }
 
-    if (httpArray.indexOf(action) == -1 && action !== 'client' && action !== 'method') {
+    if (_.contains(httpArray, action) && action !== 'client' && action !== 'method') {
```
Comment (Kestanous) [inline lib/can.js]: Same as above

File: lib/can.js [hunk 3]
```diff
@@ -61,48 +61,33 @@ Can = {
...
-        if (_.isFunction(rule.condition) && !rule.condition.apply(rule.condition, conditions)) {
-          //nothing
-        } else if (_.isFunction(rule.user)) {
-          if (rule.user.apply(rule.user, _.union([user].concat(conditions)))) {
-            passes = true;
-          }
-        } else {
-          passes = true;
-        }
+      passes = _.some(Rules[action][subject], function (rule) {
```
Comment (Kestanous) [inline lib/can.js]: Nice! I had never used `_.some`, very useful.

**Comment by Kestanous** [comment]
Well that was rather round about but I fixed and merged via #6

File: lib/can.js (hunk 1 above)
Comment (erasaur) [inline lib/can.js]: Ack I misread that as a `!=`... good catch :)

**Comment by erasaur** [comment]
Great. I think it's a good idea to have the user explicitly whitelist rules and default to deny. That way, they won't be accidentally letting something go through, e.g if they overlook a certain case.

Haven't had a chance to use this in an app yet, unfortunately. When I get the time I might work this package into one of my projects and see how we might improve it.

**Pull request merged**